content/5_Plat/Sweep_Line.mdx

```diff
@@ -55,6 +55,89 @@ To achieve the desired $\mathcal{O}(n)$ complexity per layer, we need to be able
 
 Because each step now runs in linear time and there is a total of $\lceil \log n \rceil$ steps, by the [master theorem](https://en.wikipedia.org/wiki/Master_theorem_(analysis_of_algorithms)) our solution now runs in $\mathcal{O}(n\log n)$.
 
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+struct Point {
+	long double x, y;
+	bool operator<(const Point &other) {
+		if (x == other.x) return y < other.y;
+		return x < other.x;
+	}
+};
+
+const pair<Point, Point> INF{{-1e9, -1e9}, {1e9, 1e9}};
+
+long double dist(const pair<Point, Point> &a) {
+	long double d1 = a.first.x - a.second.x;
+	long double d2 = a.first.y - a.second.y;
+	return sqrt(d1 * d1 + d2 * d2);
+}
+
+pair<Point, Point> get_closest_points(const pair<Point, Point> &a,
+                                      const pair<Point, Point> &b) {
+	return (dist(a) < dist(b) ? a : b);
+}
+
+/* Brute force for points with near mid_point */
```

Comment:
```suggestion
/** Brute force for points with near mid_point */
```
what's `mid_point`? clarify that as well

Reply:
Done. I said - `the median point in the sorted array`